festim/stepsize.py

```diff
@@ -10,19 +10,74 @@ class Stepsize:
 
     Attributes:
         initial_value (float, int): initial stepsize.
+        growth_factor (float): factor by which the stepsize is
+            increased when adapting
+        cutback_factor (float): factor by which the stepsize is
+            decreased when adapting
+        target_nb_iterations (int): number of Newton iterations
+            over (resp. under) which the stepsize is increased
+            (resp. decreased)
     """
 
     def __init__(
         self,
         initial_value,
     ) -> None:
         self.initial_value = initial_value
+        self.growth_factor = None
+        self.cutback_factor = None
+        self.target_nb_iterations = None
+
+        # TODO should this class hold the dt object used in the formulation
+
+    @property
+    def adaptive(self):
+        return self.growth_factor or self.cutback_factor or self.target_nb_iterations
+
+    @property
+    def growth_factor(self):
+        return self._growth_factor
+
+    @growth_factor.setter
+    def growth_factor(self, value):
+        if value is not None:
+            if value < 1:
+                raise ValueError("growth factor should be greater than one")
+
+        self._growth_factor = value
+
+    @property
+    def cutback_factor(self):
+        return self._cutback_factor
+
+    @cutback_factor.setter
+    def cutback_factor(self, value):
+        if value is not None:
+            if value > 1:
+                raise ValueError("cutback factor should be smaller than one")
+
+        self._cutback_factor = value
+
+    def modify_value(self, value, nb_iterations, t=None):
+        if not self.is_adapt(t):
```

Comment:
If you already have a property adaptive, why not use it here? As it only needs to be modified if the stepsize is adaptive 
```suggestion
        if not self.adaptive:
```

Reply:
Right I understand the confusion. I guess I forgot to show why/how we use `is_adapt(t)`

The idea is that sometimes you want to adapt the timestep at certain times only.

By default, it's always. But say we want to adapt if for t between 0, 100, and 200-1000, etc. then you can override the `is_adapt` method to reflect this behaviour